src/main/java/com/sap/oss/phosphor/fosstars/data/github/CodeOfConductGuidelineInfo.java

```diff
@@ -0,0 +1,205 @@
+   *
+   * @return A list of patterns that describe required content in code of conduct guidelines.
+   */
+  List<Pattern> requiredContentPatterns() {
+    return new ArrayList<>(requiredContentPatterns);
+  }
+
+  /**
+   * Set a list of patterns that describe required content in code of conduct guidelines.
+   *
+   * @param patterns The patterns.
+   * @return This data provider.
+   */
+  public CodeOfConductGuidelineInfo requiredContentPatterns(String... patterns) {
+    return requiredContentPatterns(asList(patterns));
+  }
+
+  /**
+   * Set a list of patterns that describe required content in code of conduct guidelines.
+   *
+   * @param patterns The patterns.
+   * @return This data provider.
+   */
+  public CodeOfConductGuidelineInfo requiredContentPatterns(List<String> patterns) {
+    Objects.requireNonNull(patterns, "Oops! Patterns can't be null!");
+    requiredContentPatterns.clear();
+    requiredContentPatterns.addAll(
+        patterns.stream()
+            .map(pattern -> Pattern.compile(pattern, Pattern.DOTALL)).collect(Collectors.toList()));
+    return this;
+  }
+
+  @Override
+  public Set<Feature<?>> supportedFeatures() {
+    return setOf(HAS_CODE_OF_CONDUCT, HAS_REQUIRED_TEXT_IN_CODE_OF_CONDUCT_GUIDELINE);
+  }
+
+  @Override
+  protected ValueSet fetchValuesFor(GitHubProject project) throws IOException {
+    logger.info("Gathering info about project's code of conduct guidelines ...");
+
+    Optional<String> content = lookForCodeOfConductGuideline(project);
+    if (!content.isPresent()) {
+      return ValueHashSet.from(
+        HAS_CODE_OF_CONDUCT.value(false),
+        HAS_REQUIRED_TEXT_IN_CODE_OF_CONDUCT_GUIDELINE.value(false));
+    }
+
+    ValueSet values = new ValueHashSet();
+    values.update(HAS_CODE_OF_CONDUCT.value(true));
+    values.update(infoAboutCodeOfConductGuideline(content.get()));
+
+    return values;
+  }
+
+  /**
+   * Looks for a code of conduct guideline in a project.
+   *
+   * @param project The project.
+   * @return Content of a code of conduct guideline if found.
+   * @throws IOException If something went wrong.
+   */
+  private Optional<String> lookForCodeOfConductGuideline(GitHubProject project) throws IOException {
+    LocalRepository repository = GitHubDataFetcher.localRepositoryFor(project);
+
+    for (String path : knownCodeofConductGuidelineFiles) {
+      Optional<String> content = repository.readTextFrom(path);
+      if (content.isPresent()) {
+        return content;
+      }
+    }
+
+    return Optional.empty();
+  }
+
+  /**
+   * Extract info about a code of conduct guideline.
+   *
+   * @param content Content of the code of conduct guideline.
+   * @return A value of {@link OssFeatures#HAS_REQUIRED_TEXT_IN_CODE_OF_CONDUCT_GUIDELINE}.
+   */
+  private Value<Boolean> infoAboutCodeOfConductGuideline(String content) {
+    if (requiredContentPatterns.isEmpty()) {
+      return HAS_REQUIRED_TEXT_IN_CODE_OF_CONDUCT_GUIDELINE.value(true);
+    }
```

Comment:
This is what bothered me. If the required patterns list is empty.

Why would we set the `HAS_REQUIRED_TEXT` as `true`?